common/plugins/org.jboss.tools.common.model.ui/src/org/jboss/tools/common/model/ui/views/palette/PaletteInsertHelper.java

```diff
@@ -138,6 +138,12 @@ protected void insertIntoEditorInternal(IDocument doc, Properties p, IPositionCo
 						getLineDelimiter(doc, firstLine - 1).length();
 			}
 			
+			if(start < 0){
+				start = 0;
+			}else if(start > doc.getLength()){
+				start = doc.getLength();
+			}
+			

```

Comment:
We cannot just use 0 or EOF as a position for the tag. It's not a solution if we insert the tag in some wrong place. We should find out why the position is not correct here.